include/pybind11/numpy.h

```diff
@@ -478,7 +480,7 @@ class array : public buffer {
         }
 
         auto tmp = reinterpret_steal<object>(api.PyArray_NewFromDescr_(
-            api.PyArray_Type_, descr.release().ptr(), (int) ndim, shape.data(), strides.data(),
+            api.PyArray_Type_, descr.release().ptr(), (int) ndim, shape.v.data(), strides.v.data(),
```

Comment:
I added an lvalue reference conversion operator (in addition to the rvalue operator), so I can just grab a vector reference before that and call `.data()` on it.